Manito/Manito/Screens/ParticipateRoom/ViewModel/ParticipateRoomViewModel.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  ParticipateRoomViewModel.swift
+//  Manito
+//
+//  Created by 이성호 on 2023/09/04.
+//
+
+import Combine
+import Foundation
+
+final class ParticipateRoomViewModel: ViewModelType {
+    
+    typealias Counts = (textCount: Int, maxCount: Int)
+    
+    // MARK: - property
+    
+    private let maxCount: Int = 6
+    private let roomInfoSubject = PassthroughSubject<ParticipatedRoomInfo, Error>()
+    
+    private let participateRoomService: ParticipateRoomService
+    private var cancellable = Set<AnyCancellable>()
+    
+    struct Input {
+        let viewDidLoad: AnyPublisher<Void, Never>
+        let textFieldDidChanged: AnyPublisher<String, Never>
+        let nextButtonDidTap: AnyPublisher<String, Never>
+    }
+    
+    struct Output {
+        let counts: AnyPublisher<Counts, Never>
+        let fixedTitleByMaxCount: AnyPublisher<String, Never>
+        let isEnabled: AnyPublisher<Bool, Never>
+        let roomInfo: PassthroughSubject<ParticipatedRoomInfo, Error>
+    }
+    
+    func transform(from input: Input) -> Output {
+        let countViewDidLoadType = input.viewDidLoad
+            .map { [weak self] _ -> Counts in
+                (0, self?.maxCount ?? 0)
+            }
+        
+        let countTextFieldDidChangedType = input.textFieldDidChanged
+            .map { [weak self] text -> Counts in
+                return (text.count, self?.maxCount ?? 0)
+            }
+        
+        let mergeCount = Publishers.Merge(countViewDidLoadType, countTextFieldDidChangedType)
+            .eraseToAnyPublisher()
+        
+        let fixedTitle = input.textFieldDidChanged
+            .map { [weak self] text -> String in
+                let isOverMaxCount = self?.isOverMaxCount(titleCount: text.count, maxCount: self?.maxCount ?? 0) ?? false
+                
+                if isOverMaxCount {
+                    let endIndex = text.index(text.startIndex, offsetBy: self?.maxCount ?? 0)
+                    let fixedText = text[text.startIndex..<endIndex]
+                    
+                    return String(fixedText)
+                }
+
+                return text
+            }
+            .eraseToAnyPublisher()
+        
+        let isEnabled = input.textFieldDidChanged
+            .map { text -> Bool in
+                return text.count == 6
+            }
+            .eraseToAnyPublisher()
```

Comment:
p5;

```suggestion
        let isEnabled = input.textFieldDidChanged
            .map { $0.count == 6 }
            .eraseToAnyPublisher()
```

리턴값을 명시하지 않아도 위 같이 작성할 수 있습니다!
선언형 프로그래밍의 장점이라고 생각되어 남겨봤습니다..
굳이 수정하지 않으셔도 됩니다!

Reply:
저희끼리 얘기는 안해봤지만 저희가 코드 짤 때 항상 중요시 했던게 가독성 이었던거 같아서  타입을 명시하고 생략해도 되지만 적어봤습니다.. 

이건 약어 사용에 대한 얘기였지만 포함된다고 생각했네요 후후
![스크린샷 2023-09-12 오후 4 51 39](https://github.com/DeveloperAcademy-POSTECH/MC2-Team5-Firefighter/assets/59243274/619c0241-fae4-466a-9db1-9aa1c52e9c95)

이 부분은 이따 회의 때 살짝 얘기하고 수정하겠습니다!